lib/msal-core/src/messaging/MessageListener.ts

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { WindowType, MessageHelper } from "./MessageHelper";
+import { WindowUtils } from "./../utils/WindowUtils";
+import { MessageType, PAYLOAD } from "./MessageHelper";
+import { MessageCache } from "./MessageCache";
+import { MessageDispatcher } from "./MessageDispatcher";
+import { Logger } from "./../Logger";
+import { iframeRedirectCallback } from "./../UserAgentApplication";
+
+export class MessageListener {
+
+    private messageCache: MessageCache;
+    private logger: Logger;
+    private topFrameOrigin: string;
+    private embeddedFrameOrigin: string;
+    private consentNeeded: boolean;
+    private iframeRedirectCallback: iframeRedirectCallback;
+
+    /**
+     * initialize the message listener, and register the callback to process messages
+     */
+    constructor(messageCache: MessageCache, logger: Logger, topFrameOrigin?: string, embeddedFrameOrigin?: string) {
+        this.messageCache = messageCache;
+        this.logger = logger;
+        this.topFrameOrigin = topFrameOrigin;
+        this.embeddedFrameOrigin = embeddedFrameOrigin;
+
+        // listen to all incoming messages
+        window.addEventListener("message", this.receiveMessage, false);
+    }
+
+    public setCallBack(consentNeeded: boolean, callback: iframeRedirectCallback) {
+        this.consentNeeded = consentNeeded;
```

Comment:
I dont think you actually need this boolean. The truthiess of `iframeRedirectCallback` will tell you if you should invoke it.

Reply:
Done